addons/print/models/print_printer.py

```diff
@@ -177,14 +191,53 @@ def spool(self, document, title=None, copies=1):
         return True
 
     def spool_report(self, docids, report_name, data=None, title=None, copies=1):
-        """Spool report to printer"""
+        """
+        This function will render and spool the report to the printer that it
+        is called on, as long as the printer has a report_type attribute matching that of the desired
+        report.
+        Return: boolean
+        """
         # pylint: disable=too-many-arguments, too-many-locals
 
         if copies <= 0:
             _logger.info(_("Zero or fewer copies requested, nothing will be printed."))
             return True
 
-        # Identify reports
+        reports = self._fetch_reports_for_printing(report_name)
+        required_report_types = set(reports.mapped("report_type"))
+        printers_by_report_type = self._assign_printers_to_report(required_report_types)
+        self._check_every_report_type_has_printer(printers_by_report_type, required_report_types)
+
+        # CPCL reports require number of copies passed into the template via data
+        cpcl_data = {"copies": copies}
+        if data:
+            cpcl_data.update(data)
+
+        # Generate dictionary with {report type: [(title, *rendered_document*), ...]} for each report type
+        documents = defaultdict(list)
+        for report in reports:
+            documents[report.report_type].append(
+                (
+                    ("%s %s" % (report.name, str(docids))) if title is None else title,
```

Comment:
Suggestion: Put these two values into variables, e.g. `doc_title` and `rendered_doc`

These few lines are tricky to decipher atm.

Something like this is easier to read imo.

```
doc_title = title or ("%s %s" % (report.name, str(docids)))
rendered_doc = report._render(docids, cpcl_data if report.report_type == "qweb-cpcl" else data)[0]

documents[report.report_type].append(doc_title, rendered_doc)
```